systemvm/patches/debian/config/opt/cloud/bin/cs/CsHelper.py

```diff
@@ -179,12 +179,12 @@ def get_hostname():
         return line.strip()
 
 
-def execute(command):
+def execute(command, wait=True):
     """ Execute command """
     logging.debug("Executing: %s" % command)
     p = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
-    result = p.communicate()[0]
-    return result.splitlines()
+    if wait:
+        return p.communicate()[0].splitlines()
```

Comment:
What happens if the command does not return?  It seems like there should be a wait period to timeout operations that run too long.